keras_hub/src/models/yolo_v8/yolo_v8_label_encoder.py

```diff
@@ -0,0 +1,257 @@
+                candidate_idxs[:, :, k], num_anchors
+            )
+
+        # We zero-out the overlap for anchor, GT box pairs which don't match.
+        overlaps *= anchors_matched_gt_box
+        # In cases where one anchor matches to 2 GT boxes, we pick the GT box
+        # with the highest overlap as a max.
+        gt_box_matches_per_anchor = ops.argmax(overlaps, axis=1)
+        gt_box_matches_per_anchor_mask = ops.max(overlaps, axis=1) > 0
+        # TODO(ianstenbit): Once ops.take_along_axis supports -1 in Torch,
+        # replace gt_box_matches_per_anchor with
+        # ops.where(
+        #     ops.max(overlaps, axis=1) > 0, ops.argmax(overlaps, axis=1), -1
+        # )
+        # and get rid of the manual masking
+        gt_box_matches_per_anchor = ops.cast(gt_box_matches_per_anchor, "int32")
+
+        # We select the GT boxes and labels that correspond to anchor matches.
+        bbox_labels = ops.take_along_axis(
+            gt_bboxes, gt_box_matches_per_anchor[:, :, None], axis=1
+        )
+        bbox_labels = ops.where(
+            gt_box_matches_per_anchor_mask[:, :, None], bbox_labels, -1
+        )
+        class_labels = ops.take_along_axis(
+            gt_labels, gt_box_matches_per_anchor, axis=1
+        )
+        class_labels = ops.where(
+            gt_box_matches_per_anchor_mask, class_labels, -1
+        )
+
+        class_labels = ops.one_hot(
+            ops.cast(class_labels, "int32"), self.num_classes
+        )
+
+        # Finally, we normalize an anchor's class labels based on the relative
+        # strength of the anchors match with the corresponding GT box.
+        alignment_metrics *= anchors_matched_gt_box
+        max_alignment_per_gt_box = ops.max(
+            alignment_metrics, axis=-1, keepdims=True
+        )
+        max_overlap_per_gt_box = ops.max(overlaps, axis=-1, keepdims=True)
+
+        normalized_alignment_metrics = ops.max(
+            alignment_metrics
+            * max_overlap_per_gt_box
+            / (max_alignment_per_gt_box + self.epsilon),
+            axis=-2,
+        )
+        class_labels *= normalized_alignment_metrics[:, :, None]
+
+        # On TF backend, the final "4" becomes a dynamic shape so we include
+        # this to force it to a static shape of 4. This does not actually
+        # reshape the Tensor.
+        bbox_labels = ops.reshape(bbox_labels, (-1, num_anchors, 4))
+        return (
+            ops.stop_gradient(bbox_labels),
+            ops.stop_gradient(class_labels),
+            ops.stop_gradient(
+                # ops.cast(gt_box_matches_per_anchor > -1, "float32")
+                ops.cast(gt_box_matches_per_anchor > 0, "float32")
+            ),
+        )
+
+    def call(
+        self, scores, decode_bboxes, anchors, gt_labels, gt_bboxes, gt_mask
+    ):
+        """Computes target boxes and classes for anchors.
+
+        Args:
+            scores: a Float Tensor of shape (batch_size, num_anchors,
```

Comment:
Please use backticks around code keywords, such as shape tuples. All docstrings are rendered as markdown.

Reply:
@oarriaga this needs to be addressed.